cli/cli/commands/service/update/update.go

```diff
@@ -0,0 +1,421 @@
+				filesArtifactMountpointDelimiter,
+				filesArtifactMountsDelimiter,
+				filesArtifactMountpointDelimiter,
+				command_str_consts.FilesCmdStr,
+				command_str_consts.FilesUploadCmdStr,
+			),
+			Type:    flags.FlagType_String,
+			Default: "",
+		},
+	},
+	Args: []*args.ArgConfig{
+		enclave_id_arg.NewEnclaveIdentifierArg(
+			enclaveIdentifierArgKey,
+			engineClientCtxKey,
+			isEnclaveIdArgOptional,
+			isEnclaveIdArgGreedy,
+		),
+		service_identifier_arg.NewServiceIdentifierArg(
+			serviceIdentifierArgKey,
+			enclaveIdentifierArgKey,
+			isServiceIdentifierArgGreedy,
+			isServiceIdentifierArgOptional,
+		),
+	},
+	RunFunc: run,
+}
+
+func run(
+	ctx context.Context,
+	kurtosisBackend backend_interface.KurtosisBackend,
+	_ kurtosis_engine_rpc_api_bindings.EngineServiceClient,
+	_ metrics_client.MetricsClient,
+	flags *flags.ParsedFlags,
+	args *args.ParsedArgs,
+) error {
+	enclaveIdentifier, err := args.GetNonGreedyArg(enclaveIdentifierArgKey)
+	if err != nil {
+		return stacktrace.Propagate(err, "Expected a value for non-greedy enclave identifier arg '%v' but none was found; this is a bug in the Kurtosis CLI!", enclaveIdentifierArgKey)
+	}
+
+	// TODO: need to enforce this is always service name, and not service uuid or short uuid
+	serviceName, err := args.GetNonGreedyArg(serviceIdentifierArgKey)
+	if err != nil {
+		return stacktrace.Propagate(err, "Expected a value for non-greedy enclave identifier arg '%v' but none was found; this is a bug in the Kurtosis CLI!", serviceIdentifierArgKey)
+	}
+
+	kurtosisCtx, err := kurtosis_context.NewKurtosisContextFromLocalEngine()
+	if err != nil {
+		return stacktrace.Propagate(err, "An error occurred creating Kurtosis Context from local engine")
+	}
+
+	enclaveCtx, err := kurtosisCtx.GetEnclaveContext(ctx, enclaveIdentifier)
+	if err != nil {
+		return stacktrace.Propagate(err, "An error occurred getting an enclave context from enclave info for enclave '%v'", enclaveIdentifier)
+	}
+
+	var overrideImage string
+	var overridePorts map[string]*kurtosis_core_rpc_api_bindings.Port
+	var overrideFilesArtifactsMountpoint map[string]string
+	var overrideEntrypoint []string
+	var overrideCmd []string
+	var overrideEnvVars map[string]string
+
+	imageStr, err := flags.GetString(serviceImageFlagKey)
+	if err != nil {
+		return stacktrace.Propagate(err, "An error occurred getting the image using key '%v'", serviceImageFlagKey)
+	}
+	overrideImage = imageStr
+
+	cmdStr, err := flags.GetString(cmdArgsFlagKey)
+	if err != nil {
+		return stacktrace.Propagate(err, "An error occurred getting the cmd using key '%v'", cmdArgsFlagKey)
+	}
+	if cmdStr != "" {
+		overrideCmd = []string{cmdStr}
+	}
+
+	entrypointStr, err := flags.GetString(entrypointBinaryFlagKey)
+	if err != nil {
+		return stacktrace.Propagate(err, "An error occurred getting the ENTRYPOINT binary using key '%v'", entrypointBinaryFlagKey)
```

Comment:
would be much eaiser to read if you defined a parseFlags() or something function that returned a struct with all the different overrides